### ISSUE TITLE
unittest failures throw 'object has no attribute traceback' error

### ISSUE THREAD

**Issue by Ethan-Arrowood** [brython-dev/brython#914]
Using the Brython 3.7.0rc1 in browser editor I've entered the following python code:

```python
import unittest

class UserCode():
    def add(x, y):
        return x + y
    
class TestUserCode(unittest.TestCase):
    
    def test_returnsNumber(self):
        self.assertTrue(isinstance(UserCode.add(1, 1), int))
        
    def test_addsCorrectly(self):
        self.assertEqual(UserCode.add(1, 1), 2)
        
    # testing how brython unittest responds to errors
    def test_fails(self):
        self.assertEqual(UserCode.add(2, 2), 5)

suite = unittest.TestLoader().loadTestsFromTestCase(TestUserCode)
unittest.TextTestRunner(verbosity=1).run(suite)
```

The outut (with Debug enabled) is as so:
```
<completed in  44.00 ms>
.
AttributeError: object has no attribute traceback
Traceback (most recent call last):
  module lambda5 line 2, in <lambda>
    from browser import html
  module editor line 112, in run
    exec(src, ns)
  module <module> line 19
    unittest.TextTestRunner(verbosity=1).run(suite)
  module unittest.runner line 176, in run
    test(result)
  module unittest.suite line 84, in __call__
    return self.run(*args,**kwds)
  module unittest.suite line 122, in run
    test(result)
  module unittest.case line 663, in __call__
    return self.run(*args,**kwds)
  module unittest.case line 615, in run
    testMethod()
  module contextlib line 158, in __exit__
    if sys.exc_info()[1]is value:
```

Any idea what is going on here?

**Comment by PierreQuentel** [comment]
@Ethan-Arrowood Thanks for the report :-)

**Comment by Ethan-Arrowood** [comment]
Thank you Pierre! If you wouldn’t mind, can you explain why this wasn’t working and what you did to fix it? (This is just for my own personal knowedge growth) thank you!

**Comment by PierreQuentel** [comment]
@Ethan-Arrowood 
If you click on the link in the message "PierreQuentel closed this in f4048ba 6 days ago" above, you have the list of modified scripts. You can ignore __brython.js__, __brython_dist.js__ and __version_info.js__ (these are scripts generated by scripts/make_dist.py each time a core Brython script is modified) and the test script.

The real change is in __py_generator.js__ : the line
```javascript
    if(_b_.isinstance(value, _b_.type)){value = value()}
```
has been replaced by
```javascript
    if(_b_.isinstance(value, _b_.type)){value = $B.$call(value)()}
```
This code is run if an exception is thrown to a generator by `generator.throw(exception)`. In Python, `exception` can be either an exception _class_ or an exception _instance_. Brython transforms an exception classe into an exception instance.

`_b_.isinstance(value, _b_.type)`  is true if `value` is a class. If it is the case, the code to create an instance is not `value = value()` (this was true in old Brython versions) but `value = $B.$call(value)` ; `$B.$call` is defined in __py_utils.js__ and is used for calls : each Python code `callable(...)` is transformed into `$B.$call(callable)(...)`.

**Comment by Ethan-Arrowood** [comment]
Thank you for this great explanation! I appreciate it a lot 😄
